### PR TITLE
Advertising Event

### PR DESCRIPTION
**Pull request by fokinv**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

### CODE REVIEW

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -33,6 +33,7 @@ const CHARACTERISTIC_ERROR: &'static str = "No characteristic found";
 const DESCRIPTOR_ERROR: &'static str = "No descriptor found";
 const VALUE_ERROR: &'static str = "No characteristic or descriptor found with that id";
 const SECURITY_ERROR: &'static str = "The operation is insecure";
+const NOT_SUPPORTED: &'static str = "This feature is not supported yet";
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:36]: `NOT_SUPPORTED_ERROR`

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -166,6 +167,12 @@ impl BluetoothManager {
                 BluetoothMethodMsg::RequestDevice(options, sender) => {
                     self.request_device(options, sender)
                 },
+                BluetoothMethodMsg::WatchAdvertisements(id, sender) => {
+                    self.watch_advertising(id, sender)
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:171]: why is this `watch_advertising` and not `watch_advertisements`, like above?

File: components/script/dom/bluetoothadvertisingevent.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            device: MutHeap::new(device),
+            name: name,
+            appearance: appearance,
+            txPower: txPower,
+            rssi: rssi,
+        }
+    }
+
+    pub fn new(global: GlobalRef,
+               type_: Atom,
```
Comment (mmatyas) [inline components/script/dom/bluetoothadvertisingevent.rs:48]: `type_`? be more creative please :)

File: components/script/dom/bluetoothadvertisingevent.rs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+               rssi: Option<i8>)
+               -> Root<BluetoothAdvertisingEvent> {
+        let event = box BluetoothAdvertisingEvent::new_inherited(device, name, appearance, txPower, rssi);
+        let ev = reflect_dom_object(event, global, BluetoothAdvertisingEventBinding::Wrap);
+        {
+            let event = ev.upcast::<Event>();
+            event.init_event(type_,
+                             bool::from(bubbles),
+                             bool::from(cancelable));
+        }
```
Comment (mmatyas) [inline components/script/dom/bluetoothadvertisingevent.rs:64]: why `{}`?

File: components/script/dom/bluetoothadvertisingevent.rs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+               type_: Atom,
+               bubbles: EventBubbles,
+               cancelable: EventCancelable,
+               device: &BluetoothDevice,
+               name: Option<DOMString>,
+               appearance: Option<u16>,
+               txPower: Option<i8>,
+               rssi: Option<i8>)
+               -> Root<BluetoothAdvertisingEvent> {
+        let event = box BluetoothAdvertisingEvent::new_inherited(device, name, appearance, txPower, rssi);
```
Comment (mmatyas) [inline components/script/dom/bluetoothadvertisingevent.rs:57]: is this `event` even used?

File: components/script/dom/bluetoothdevice.rs [hunk 6]
```diff
@@ -4,48 +4,57 @@
...
+use dom::bindings::js::{JS, Root, MutNullableHeap};
 use dom::bindings::reflector::{Reflectable, Reflector, reflect_dom_object};
 use dom::bindings::str::DOMString;
-use dom::bluetoothadvertisingdata::BluetoothAdvertisingData;
 use dom::bluetoothremotegattserver::BluetoothRemoteGATTServer;
+use ipc_channel::ipc::{self, IpcSender};
+use net_traits::bluetooth_thread::BluetoothMethodMsg;
+use std::cell::Cell;
+
+const WRONG_FUNC_CALL: &'static str = "The device already (un)watches the Advertisements.";
```
Comment (mmatyas) [inline components/script/dom/bluetoothdevice.rs:18]: This message sounds weird to me. I'd prefer watch/unwatch separated.

File: components/script/dom/bluetoothdevice.rs [hunk 7]
```diff
@@ -59,13 +68,53 @@ impl BluetoothDeviceMethods for BluetoothDevice {
...
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothdevice-watchadvertisements
+    fn WatchAdvertisements(&self) -> ErrorResult {
+        if self.watchingAdvertisements.get() {
+            return Err(Type(WRONG_FUNC_CALL.to_owned()));
+        }
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::WatchAdvertisements(self.id.to_string(), sender)).unwrap();
+        let result = receiver.recv().unwrap();
```
Comment (mmatyas) [inline components/script/dom/bluetoothdevice.rs:84]: add a new line somewhere around here

File: components/script/dom/bluetoothdevice.rs [hunk 8]
```diff
@@ -59,13 +68,53 @@ impl BluetoothDeviceMethods for BluetoothDevice {
...
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothdevice-unwatchadvertisements
+    fn UnwatchAdvertisements(&self) -> ErrorResult {
+        if !self.watchingAdvertisements.get() {
+            return Err(Type(WRONG_FUNC_CALL.to_owned()));
+        }
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.get_bluetooth_thread().send(
+            BluetoothMethodMsg::UnwatchAdvertisements(self.id.to_string(), sender)).unwrap();
+        let result = receiver.recv().unwrap();
```
Comment (mmatyas) [inline components/script/dom/bluetoothdevice.rs:104]: here too

File: components/script/dom/bluetoothdevice.rs (hunk 7 above)
Comment (fokinv) [inline components/script/dom/bluetoothdevice.rs:84]: There isn't any empty line in any of the functions in the DOM side. Wouldn't it be better to do this with every file later in a style fix, or something?

**Pull request closed without merging**